test/blackbox-tests/test-cases/dune-package/run.t

```diff
@@ -28,3 +28,75 @@
     (main_module_name C)
     (modules ((name Y) (obj_name c__Y) (visibility public) (impl)))
     (wrapped true)))
+
+Build with "--store-orig-source-dir" profile
+  $ dune build --store-orig-source-dir
+  $ cat _build/install/default/lib/a/dune-package
```

Comment:
I'd put `| grep -A 1 orig_src_dir`, just to focus the test on the relevant part. Otherwise the reader has to manually scan the output to check that the test is successful.